### ISSUE TITLE
Architecture Family support

### ISSUE THREAD

**Issue by GeorgeHahn** [CraneStation/target-lexicon#9]
I would like to add support for listing architecture family. This would be a mapping from `Architecture` to a smaller set of `ArchitectureFamily` values - ARM, x86, Risc-V, and so on.

Do you think this is a good idea?

**Comment by sunfishcode** [comment]
Yeah, that makes sense. Would armv7 and aarch64 would be different families?

**Comment by GeorgeHahn** [comment]
That's a good question; I'm not sure I have an informed opinion here. I wonder if this would still be valuable alongside #11?

I'll review triple usage in `cc` tomorrow to see if it adds some clarity.

**Comment by sunfishcode** [comment]
#11 added this for ARM and AArch64, and #42 adds this for Riscv32, Riscv65, and X86.
